Lib/http/server.py

```diff
@@ -686,12 +687,42 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
         try:
+            fs = os.fstat(f.fileno())
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
+                    and "If-None-Match" not in self.headers):
+                # compare If-Modified-Since and time of last file modification
+                try:
+                    ims = email.utils.parsedate_to_datetime(
+                        self.headers["If-Modified-Since"])
+                except:
+                    # ignore ill-formed values
+                    ims = None
+                if ims is not None:
```

Comment:
`email.utils.parsedate_to_datetime()` either returns a `datetime.datetime()` or raises an exception, in which case `ims` is set to `None`.